### DISCUSSION TITLE
globalStyle support nesting

### DISCUSSION THREAD

**Discussion by Dan503** [vanilla-extract-css/vanilla-extract#174] [Ideas]
I would really like to be able to do this:

```js
globalStyle('button', {
	backgroundColor: 'black',
	color: 'white'
	'&:hover': {
		backgroundColor: 'white',
		color: 'black'
	}
});
```

but I get the error:

> ''&:hover'' does not exist in type 'GlobalStyleRule'

**Comment by mattcompiles** [comment]
Hey Bud. This is something we've chatted about in theory but don't support for now. My main issues with it are as follows:

- Simple cases are easy enough, but you can pass any selector you like to `globalStyle` and we'd need a way to merge the styles together. It's probably doable but it's not a simple addition. e.g. 
```ts
globalStyle('.someClass button, .someOtherClass:not(button):active', {
	backgroundColor: 'black',
	color: 'white'
	':hover': {
		backgroundColor: 'white',
		color: 'black'
	}
});
```
- `globalStyle` is a good escape hatch but it's not the suggested API for most use-cases. I think you lose a lot of what is good about vanilla-extract if you use it exclusively. So hopefully its use is minimal enough that this doesn't become a huge issue.

**Comment by marcingajdatsh** [comment]
A temporary workaround for this is to create a second global style:
```
globalStyle('a', {
  color: 'blue',
});

globalStyle('a:hover', {
  color: 'lightblue',
});
```

**Reply by ubmit** [reply]
> A temporary workaround for this is to create a second global style:
> 
> ```
> globalStyle('a', {
>   color: 'blue',
> });
> 
> globalStyle('a:hover', {
>   color: 'lightblue',
> });
> ```

Thanks for the workaround 🙏

**Reply by debitan** [reply]
Thanks. Migrating from a CSS Modules that has some nested global styles like this and good to be able to move them across to vanilla-extract for the time being.

**Comment by sami616** [comment]
Would be such a great addition to be able to use the `selectors` property here, i know that `globalStyle` isn't  recommended for normal usage but it's often very necessary as an escape hatch for styling third party components. 👍🏼👍🏼👍🏼👍🏼

ie

```
globalStyle('a', {
  color: 'blue',
  selectors: { '&:hover': { color: 'red' } },
});

```

**Reply by Dan503** [reply]
It doesn't really solve the problem that mattcompiles talks about in this comment:
https://github.com/seek-oss/vanilla-extract/discussions/174#discussioncomment-837594

Also I would prefer writing just `&:hover` rather than `selectors: { '&:hover': { } }`

**Reply by sami616** [reply]
Fair. I'm quite enjoying the selectors property tbh. Using it forces me to keep all my selectors for an element together. Why not have both options? Would make the api more consistent with 'style'
